pkg/container/templates/go.tmpl

```diff
@@ -37,18 +26,78 @@ ENV CGO_ENABLED=0 \
     GOARCH=amd64 \
     GO111MODULE=on
 
+# Set working directory
+WORKDIR /build
+
 {{if .IsLocalPath}}
 # Copy the local source code
-COPY . /app/
+COPY . /build/
+
+# Build the application
+RUN go build -o /app/mcp-server {{.MCPPackage}}
+{{else}}
+# Pre-install the Go package at build time
+# This downloads all dependencies and builds the binary
+# Check if the package already has a version specifier
+RUN package="{{.MCPPackage}}"; \
+    # If package doesn't have @ version specifier, add @latest
+    if ! echo "$package" | grep -q '@'; then \
+        package="${package}@latest"; \
+    fi; \
+    # Create the app directory first
+    mkdir -p /app && \
+    # Install the package
+    go install "$package" && \
+    # Move the installed binary to a known location
+    mv $GOPATH/bin/* /app/mcp-server 2>/dev/null || \
+    # If the package name differs from binary name, try to find it
+    find $GOPATH/bin -type f -executable -exec mv {} /app/mcp-server \; 2>/dev/null || \
+    # As a fallback, build it directly (strip version for go build)
+    (base_package=$(echo "$package" | sed 's/@.*//'); \
+     go get "$package" && go build -o /app/mcp-server "$base_package")
+{{end}}
+
+# Final stage - minimal runtime image
+FROM alpine:3.20
+
+{{if .CACertContent}}
```

Comment:
why cacert is copied 2 times in the same template?

Reply:
When building in a network isolated environment, you want the copy to happen at each stage of the multi stage build. This way, the package managers can actually reach out